ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -501,19 +491,53 @@ private[appendonlydao] final class TransactionsReader(
       loggingContext: LoggingContext
   ): Source[EventsTable.Entry[E], NotUsed] =
     PaginatingAsyncStream.streamFrom(range, getNextPageRange) { range1 =>
-      if (EventsRange.isEmpty(range1))
-        Future.successful(Vector.empty)
-      else {
-        val rawEvents: Future[Vector[EventsTable.Entry[Raw[E]]]] =
-          dispatcher.executeSql(queryMetric)(query(range1))
-        rawEvents.flatMap(es =>
-          Timed.future(
-            future = Future.traverse(es)(deserializeEntry(verbose)),
-            timer = queryMetric.translationTimer,
-          )
-        )
+      fetchEventsInRange(verbose, queryMetric, query)(range1)
+    }
+
+  private def streamEventsWithFixedPageSize[E](
+      verbose: Boolean,
+      queryMetric: DatabaseMetrics,
+      query: EventsRange[Long] => Connection => Vector[
+        EventsTable.Entry[Raw[E]]
+      ],
+      fullRange: EventsRange[(Offset, Long)],
+  )(implicit loggingContext: LoggingContext): Source[EventsTable.Entry[E], NotUsed] =
+    Source
+      .fromIterator(() => Iterator.iterate(fullRange.startExclusive._2)(_ + pageSize))
+      .sliding(2)
+      .map { case Seq(startExclusive, endInclusive) =>
+        EventsRange(startExclusive, endInclusive.min(fullRange.endInclusive._2))
       }
+      .takeWhile(_.endInclusive < fullRange.endInclusive._2, inclusive = true)
+      .mapAsync(5) { eventsRange =>
+        fetchEventsInRange(
```

Comment:
this flavor of paginated streams does something completely different as the semantics of these ranges go.
- this: a sequential partition of the complete range
- that: a sequential shrinking of the partition from the from part, based on the resultsets
(is that right? do I miss something?)
Yet the actual query implementation (fetchEventsInRange) seem to serve them both. Is that right? Is that intended?